packages/block-library/src/social-links/editor.scss

```diff
@@ -114,3 +83,9 @@
 	// Windows High Contrast mode will show this outline, but not the box-shadow.
 	outline: 2px solid transparent;
 }
+
+// To ensure a better selection footprint when editing, attach the margin to the block container.
```

Comment:
This CSS should become dead eventually, but in the mean time it improves the experience.